src/config.rs

```diff
@@ -1959,6 +1959,32 @@ mod readpool_tests {
     }
 }
 
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug, Configuration)]
+#[serde(default)]
+#[serde(rename_all = "kebab-case")]
+pub struct BackupConfig {
+    pub num_threads: usize,
+}
+
+impl BackupConfig {
+    pub fn validate(&self) -> Result<(), Box<dyn Error>> {
+        if self.num_threads == 0 {
+            return Err("backup.num_threads cannot be 0".into());
+        }
+        Ok(())
+    }
+}
+
+impl Default for BackupConfig {
+    fn default() -> Self {
+        let cpu_num = SysQuota::new().cpu_cores_quota();
+        Self {
+            // use at most 75% of vCPU by default
+            num_threads: (cpu_num - cpu_num / 4).max(1),
```

Comment:
Could it be a hard limit? TiKV spawns `min(req.concurrency, 75%*cpu_num)` threads for backup. I'm worried about spawning too many threads affect online traffic for servers that have many cores.

Reply:
Changed to clamp at num-threads=32. 

But I don't see any benefits still having `req.concurrency`. Together with pingcap/br#396, this means we are going to have two concurrency axes (BR's range concurrency and TiKV's file concurrency `req.concurrency`). It could be confusing if we expose a new concurrency parameter.

If instead we use a fixed value for `req.concurrency`, it would be equivalent to the current solution. And since BR doesn't know the TiKV machine's spec, it either needs to choose a very small number (4, currently) which is slow or a very large number which breaks older TiKVs.

Users can dynamically change the `backup.num-threads` setting in any case.